src/v/raft/tests/basic_raft_fixture_test.cc

```diff
@@ -74,6 +74,30 @@ TEST_F(raft_fixture, test_empty_writes) {
       replicate(std::move(reader)).get(), "Assert failure.+Empty batch");
 }
 
+TEST_F_CORO(raft_fixture, test_stuck_append_entries) {
+    co_await create_simple_group(3);
+    auto leader = co_await wait_for_leader(10s);
+
+    for (auto& [_, node] : nodes()) {
+        node->on_dispatch([](model::node_id, raft::msg_type t) {
+            if (t == raft::msg_type::append_entries) {
+                return ss::sleep(5s);
```

Comment:
nit: maybe we could sleep for a shorter (but still sufficient) time to reduce the test run time.